### PR TITLE
feat(hadolint): implement DL4006 pipefail rule

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

- Implement DL4006 rule: warns when `RUN` has a pipe (`|`) but no prior `SHELL` with `-o pipefail`
- Track SHELL state per-instruction within each stage (resets on `FROM`), correctly handling multiple `SHELL` changes mid-stage
- Validate pipefail only with shells that support it (bash/zsh/ash — rejects `/bin/sh`)
- Skip non-POSIX shells (PowerShell, cmd.exe) entirely
- Auto-fix inserts `SHELL ["/bin/bash", "-o", "pipefail", "-c"]` before violating RUN (safety: `FixSuggestion`)
- When `tally/prefer-run-heredoc` is enabled and the command is a heredoc candidate, suppress the fix to avoid conflicting suggestions
- Fix `VariantFromShell` to strip `.exe` suffix so `powershell.exe` is correctly recognized as non-POSIX

## Test plan

- [x] All 13 original Hadolint DL4006Spec test cases ported and passing
- [x] Additional edge cases: multiline RUN, subshell pipes, `&&` chains with pipes, exec form, multiple violations per stage, state persistence/reset across stages
- [x] Fix generation tests: SHELL insertion, exec form skip, heredoc coordination
- [x] Integration tests: check + fix snapshots
- [x] `make lint` clean
- [x] Full test suite passes (`go test ./...`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added Hadolint DL4006 to detect RUNs with pipes missing pipefail and offer automatic fixes; heredoc formatting can inject pipefail and rules coordinate fixes across heuristics.

* **Documentation**
  * Updated supported Hadolint rules counts to include DL4006.

* **Tests**
  * Added comprehensive tests and updated snapshots covering DL4006, heredoc interactions, and diagnostics.

* **Chores**
  * Adjusted VS Code extension test expectations for diagnostics.

### CODE REVIEW

File: internal/rules/hadolint/dl4006.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+// determineFixShell picks the shell path to use in the SHELL fix instruction.
+func (r *DL4006Rule) determineFixShell(input rules.LintInput, stageIdx int) string {
+	fixShell := "/bin/bash"
+	if sem, ok := input.Semantic.(*semantic.Model); ok {
+		if info := sem.StageInfo(stageIdx); info != nil && len(info.ShellSetting.Shell) > 0 {
+			shellBase := strings.ToLower(path.Base(info.ShellSetting.Shell[0]))
+			if pipefailValidShells[shellBase] {
+				fixShell = info.ShellSetting.Shell[0]
+			}
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/hadolint/dl4006.go]: ✅ Addressed in 482cb67. Thanks @coderabbitai[bot]\!

**Comment by tinovyatkin** [comment]
@codex review
